miner/worker.go

```diff
@@ -360,13 +368,45 @@ func (miner *Miner) applyTransaction(env *environment, tx *types.Transaction) (*
 		gp   = env.gasPool.Gas()
 	)
 	receipt, err := core.ApplyTransaction(miner.chainConfig, miner.chain, &env.coinbase, env.gasPool, env.state, env.header, tx, &env.header.GasUsed, vm.Config{})
+	// If successful, and not just reproducing the block, check the interop executing messages.
+	if err == nil && !env.noTxs && miner.chain.Config().IsInterop(env.header.Time) {
+		// Whenever there are `noTxs` it means we are building a block from pre-determined txs. There are two cases:
+		//	(1) it's derived from L1, and will be verified asynchronously by the op-node.
+		//	(2) it is a deposits-only empty-block by the sequencer, in which case there are no interop-txs to verify (as deposits do not emit any).
+		err = miner.checkInterop(env.rpcCtx, tx, receipt)
+	}
 	if err != nil {
 		env.state.RevertToSnapshot(snap)
 		env.gasPool.SetGas(gp)
 	}
 	return receipt, err
 }
 
+func (miner *Miner) checkInterop(ctx context.Context, tx *types.Transaction, receipt *types.Receipt) error {
```

Comment:
In the big switch over error cases when building the block (call to `miner.commitTransaction`), this will result in the `default` case being hit. Could be fine to encapsulate the diff to just this area but could also have a specific error case that represents failed interop message checking

Reply:
done, good suggestion